src/suite/integration.go

```diff
@@ -0,0 +1,119 @@
+package suite
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/ilyakaznacheev/cleanenv"
+	"github.com/stretchr/testify/suite"
+	"github.com/yagobatista/taco-go-web-framework/src/database"
+	"github.com/yagobatista/taco-go-web-framework/src/server"
+)
+
+type IntegrationSuite struct {
+	suite.Suite
+
+	mu sync.Mutex
+
+	tables []any
+
+	server       *server.Server
+	serverConfig server.ServerConfig
+
+	Ctx context.Context
+
+	Client Client
+}
+
+func (this *IntegrationSuite) SetupSuite() {
+	this.loadEnv()
+	this.createDB()
+	this.migrateDB()
+}
+
+func (this *IntegrationSuite) SetupTest() {
+	this.server = server.NewServer(this.serverConfig)
+
+	this.Client = NewClient(this.server, &this.Suite)
+
+	conn := this.server.GetConnection().GetConnection()
+
+	this.Ctx = context.Background()
+
+	tx := conn.WithContext(this.Ctx).Begin()
+
+	this.Ctx = database.SetConnectionToCtx(this.Ctx, tx)
+}
+
+func (this *IntegrationSuite) RollbackTest() {
+	conn := database.GetConnectionFromCtx(this.Ctx)
+	if conn == nil {
+		return
+	}
+
+	conn.Rollback()
+}
+
+func (this *IntegrationSuite) SetServerConfig(config server.ServerConfig) {
+	this.serverConfig.Handlers = config.Handlers
+	this.serverConfig.MainMiddlewares = config.MainMiddlewares
+	this.serverConfig.Routes = config.Routes
+}
+
+func (this *IntegrationSuite) SetTables(tables []any) {
+	this.tables = tables
+}
+
+func (this *IntegrationSuite) loadEnv() {
+	err := cleanenv.ReadConfig("../.test.env", &this.serverConfig)
+	if err == nil {
+		return
+	}
+
+	err = cleanenv.ReadConfig("../../.test.env", &this.serverConfig)
+	if err == nil {
+		return
+	}
```

Comment:
duplicado? Não tem aquela função de puxar a config da env com path relativo?

Reply:
Se o problema for porque a suite abre em lugares diferente do código, podemos tentar algo desse tipo: 

`func GetEnvPath(env string) string {
	path := fmt.Sprintf(".%s.env", env)
	wdPath, _ := os.Getwd()

	splitWd := strings.Split(wdPath, "/src")
	countSlash := strings.Count(splitWd[1], "/")
	for i := 0; i < countSlash; i++ {
		path = "../" + path
	}

	return path
}`

Isso ⬆️ normaliza o caminho, sempre retornando com o numero de "../" apropriado de acordo com a profundidade relativa a "src/". Se estiver em taco/src/tests/suite esse código vai fazer "taco/src/tests/suite/../../.local.env"

Reply:
honestamente nao me lembro